frontend/src/app/features/details/parcelDescriptions/parcelDescriptions.tsx

```diff
@@ -1,59 +1,74 @@
 import { useDispatch, useSelector } from 'react-redux';
 import { AppDispatch, RootState } from '../../../Store';
 import {
   fetchParcelDescriptions,
-  IFetchParcelDescriptionParams,
-  IParcelDescriptionState,
-  setCurrentPage,
-  setResultsPerPage,
-  setSearchParam,
-  setSortBy,
-  setSortByDir,
-  setSortByInputValue,
+  initialParcelDescriptionsState,
 } from './parcelDescriptionsSlice';
 import { columns } from './parcelDescriptionsConfig';
 import { useParams } from 'react-router-dom';
 import ParcelDescriptionTable from './ParcelDescriptionTable';
+import { IParcelDescriptionsState } from './parcelDescriptionsInterfaces';
+import { RequestStatus } from '../../../helpers/requests/status';
 
 const ParcelDescriptions = () => {
   const dispatch = useDispatch<AppDispatch>();
-  const parcelDescriptionsState: IParcelDescriptionState = useSelector(
-    (state: RootState) => state.parcelDescriptions,
-  );
+  let {
+    siteId,
+    data,
+    requestStatus,
+    totalResults,
+    currentPage,
+    resultsPerPage,
+    searchParam,
+    sortBy,
+    sortByDir,
+    sortByInputValue,
+    needsUpdate,
+  } = useSelector((state: RootState) => state.parcelDescriptions);
   const { id } = useParams();
-  const siteId = Number(id);
-
-  const data = parcelDescriptionsState.data;
-  const requestStatus = parcelDescriptionsState.requestStatus;
-  const totalResults = parcelDescriptionsState.totalResults;
-  const currentPage = parcelDescriptionsState.currentPage;
-  const resultsPerPage = parcelDescriptionsState.resultsPerPage;
-  const searchParam = parcelDescriptionsState.searchParam;
-  const sortBy = parcelDescriptionsState.sortBy;
-  const sortByDir = parcelDescriptionsState.sortByDir;
-  const sortByInputValue = parcelDescriptionsState.sortByInputValue;
+  const currentSiteId = Number(id);
+  if (siteId !== currentSiteId) {
+    // The redux cache has data from another site. Re-initialize everything..
+    siteId = initialParcelDescriptionsState.siteId;
+    data = initialParcelDescriptionsState.data;
+    requestStatus = initialParcelDescriptionsState.requestStatus;
+    totalResults = initialParcelDescriptionsState.totalResults;
+    currentPage = initialParcelDescriptionsState.currentPage;
+    resultsPerPage = initialParcelDescriptionsState.resultsPerPage;
+    searchParam = initialParcelDescriptionsState.searchParam;
+    sortBy = initialParcelDescriptionsState.sortBy;
+    sortByDir = initialParcelDescriptionsState.sortByDir;
+    sortByInputValue = initialParcelDescriptionsState.sortByInputValue;
+    needsUpdate = initialParcelDescriptionsState.needsUpdate;
+  }
 
   const handleSelectPage = (newPage: number) => {
-    dispatch(setCurrentPage(newPage));
+    currentPage = newPage;
+    needsUpdate = true;
+    fetchNewParcelDescriptions();
   };
 
   const handleChangeResultsPerPage = (newResultsPerPage: number) => {
-    dispatch(setResultsPerPage(newResultsPerPage));
+    resultsPerPage = newResultsPerPage;
```

Comment:
we should manage and track state changes react way using useState() otherwise will lead to bugs and unpredictable behaviors.